module/onboarding/Ping/src/Ping.cpp

```diff
@@ -25,7 +23,7 @@ namespace module::onboarding {
 
         on<Trigger<Pong>>().then([this](const Pong& pong_msg) {
             auto ping_msg = std::make_unique<Ping>();
-            ping_msg->count = pongcount++; // TODO: Assign counter value.
+            ping_msg->count = pong_msg.count + 1; // Assign counter value.
```

Comment:
pong_msg is declared as const &, so you've promised to not touch that argument but this line is altering the state of that message. This probably wont compile. Even if it did, it still is wrong. You should be creating a new message and emitting that one instead.